CONTRIBUTING.md

```diff
@@ -2,7 +2,103 @@
+
+For any significant new functionality, contact the ICU development team through the icu-design mailing list first, and discuss the features, design and scope of the possible contribution. This helps ensure that the contribution is expected and will be welcome, that it will fit in well with the rest of ICU, and that it does not overlap with other development work that may be underway.
+
+While you are considering contributing code to ICU, make sure that the legal terms (see [License](#license) above) are acceptable to you and your organization.
+
+Here are several things to keep in mind when developing a potential contribution to the ICU project:
+
+1.  ICU has both C/C++ and Java versions. If you develop in one programming language, please either provide a port or make sure that the logic is clear enough so that the code can be reasonably ported. We cannot guarantee that we will port a contribution to the other library.
+
+2.  Before implementation, read and understand ICU's [coding guidelines][coding-guidelines]. Contributions that require too much adaptation to be included in the ICU tree will probably wait for a long time.
+
+3.  During implementation, try to mimic the style already present in the ICU source code.
+
+4.  Always develop the code as an integral part of the library, rather than an add-on.
+
+5.  Always provide enough test code and test cases. We require that our APIs are 100% tested and that tests cover at least 85% of the ICU library code. Make sure that your tests are integrated into one of ICU's test suites ([cintltst][cintltst] and [intltest][intltest] for ICU4C and [com.ibm.icu.dev.test][com.ibm.icu.dev.test] classes in ICU4J). New tests and the complete test suite should pass.
+
+6.  Compile using the strictest compiler options. Due to ICU's multi-platform nature, warnings on some platforms may mean disastrous errors on other platforms. This can be enabled for C++ by using the `--enable-strict` configure option on any platform using the gcc or clang compilers.
+
+7.  Test on more than one platform. For ICU4C, it is good to combine testing on Windows with testing on Linux, Mac OS X or another Unix platform. It is always good to try to mix big and little endian platforms. For ICU4J, test using both Oracle's and IBM's JDKs and/or on Android.
+
+8.  Each contribution should contain everything that will allow building, testing and running ICU with the contribution. This usually includes: source code, build files and test files.
+
+
+## Team
+
+We have several [mailing lists][contacts]. Contributors should at least subscribe to the **icu-design** mailing list, and we also have a team-internal list that you should get added to.
+
+We meet once a week by phone. See the [agenda & meeting minutes][meetings].
+
+## Understand ICU
+
+
+*   Home page: <https://icu.unicode.org>
+
+*   User Guide: <https://unicode-org.github.io/icu/userguide/>
```

Comment:
```suggestion
*   Home page: https://icu.unicode.org

*   User Guide: https://unicode-org.github.io/icu/userguide/
```

Reply:
I can do that, just curious about the reason?  Spec:

- [URI Autolink](https://github.github.com/gfm/#uri-autolink) is what I'm using, which is a core part of Markdown
- GFM uses the [`autolink`](https://github.github.com/gfm/#uri-autolink) extension which has more implementation-dependent aspects as to when it is supported.

URIs within `<>` also have a long tradition in plaintext.  `https://…` linkifies in many but not all environments that will read and process the markdown:  think about an IDE previewing this file as part of the user's code browsing experience.

Reply:
I don't feel super strongly about these, but mostly these days URLs autolink without extra syntax, and the customary URL highlighting has largely replaced the old `<>` highlighting.

If these don't autolink here without the `<>` or if you feel strongly, then keep them as you proposed.

Reply:
It works in a lot of places, but it is out of core spec.

Reply:
Again, I don't care strongly, but I see a lot of Markdown inside Google and elsewhere, and not a lot of `<>`.